### ISSUE TITLE
Creating in-memory image

### ISSUE THREAD

**Issue by croese** [kelvin13/png#15]
I'm looking to generate PNG images in memory rather than loading from a file - is there a way to construct an image of a given size with, for example, all pixels initialized to a given color?  I don't see anything obvious in the api file but I wanted to see if I was missing something.

Thank you for creating this library.

**Comment by kelvin13** [comment]
(sorry for the late response, i didn’t get a notification)

this is described in one of the tutorials [here](https://github.com/kelvin13/png/blob/master/doc/3.0.0/tutorials.md) (the [`sepia.swift`](https://github.com/kelvin13/png/blob/master/examples/sepia.swift) one). Just conform your in-memory destination buffer type to:

```
protocol DataDestination
{
    mutating
    func write(_ buffer:[UInt8]) -> Void?
}
```

and use the `compress<Destination>(to:chunkSize:level:)` function rather than the `compress<Destination>(path:chunkSize:level:)` function.

**Comment by kelvin13** [comment]
Resolved in 4.0.0 release (see [in-memory image tutorial](https://github.com/kelvin13/png/tree/master/examples#using-in-memory-images))
